src/window.rs

```diff
@@ -251,6 +251,45 @@ impl WindowWrapper {
     }
 }
 
+#[derive(Clone)]
+struct WindowSettings {
+    refresh_rate: u64,
+    no_idle: bool,
+}
+
+fn parse_changed_setting(name: &str, value: Option<Value>) -> Value {
+    let mut settings = SETTINGS.get::<WindowSettings>();
+    match name {
+        "refresh_rate" => {
+            if let Some(value) = value {
+                settings.refresh_rate = value.as_u64().unwrap(); // TODO -- handle wrong data type
+                SETTINGS.set(&settings);
```

Comment:
You mentioned this above, that the listeners are in charge of setting and parsing the default values. This feels extra weird to me because the purpose of the listener doesn't flow from the type of the listener. Just reading the settings code, it isn't immediately obvious that the listener should also set the current value.

Unfortunately, I don't have a suggestion for how to change this to make it better... I'll think some more.